### ISSUE TITLE
Rollins Department/subfield names aren't the same between My Program tab and Submit tab

### ISSUE THREAD

**Issue by RachelDCE** [curationexperts/laevigata#1748]
Department/subfield names aren't the same between My Program tab and Submit tab:

My Prog Tab: Biostatistics and Bioinformatics
Submit Tab shows: Biostatistics

My Prog Tab: Behavioral Sciences and Health Education 
Submit Tab shows: Behavioral Sciences & Health Education - MPH

My Prog Tab: Health Policy and Management - MSPH 
Submit Tab shows: Health Policy and Management

My Program Tab: Hubert Department of Global Health - MPH
Submit Tab shows: Hubert Department of Global Health

These matched:
Environmental Health
Epidemiology
Executive Masters of Public Health - MPH

_All subfields haven't been tested, but as seen below some aren't matching._ 

Example:

My Program Tab

![screen shot 2018-08-23 at 10 54 45 am](https://user-images.githubusercontent.com/23509160/44546409-71add100-a6dd-11e8-9c6c-85254d844d9f.png)

Submit Tab

![screen shot 2018-08-23 at 10 55 05 am](https://user-images.githubusercontent.com/23509160/44546422-7bcfcf80-a6dd-11e8-98ba-39af0e7cb012.png)

**Comment by RachelDCE** [comment]
I've worked with Mark on this and we have found that the Department and Subfield are correct in the dropdowns on the MyProgram tab.  The Submit tab/review is displaying the ID instead of the term - ie. Department should say Biostatistics and Informatics. Subfield should say Public Health Informatics - MSPH.

Mark looked this ETD up on the server and this is what is stored. 
```
id: 7p88cg57g
   school: ["Rollins School of Public Health"] #<ActiveTriples::Relation:0x000000000a762960>
   department: ["Behavioral Sciences and Health Education"] #<ActiveTriples::Relation:0x000000000a770380>
   subfield: ["Public Health Informatics"] #<ActiveTriples::Relation:0x000000000a760ae8>
```

**Comment by jenlindner** [comment]
There are two functions now in the code that get the term from the id, fixing these bugs, so this code just needs to be QA'ed again. I did find another bug that could be triggered by saving a subfield and then saving the tab without one, and navigating to the submit tab, where the Program tab would not display due to js errors trying to reference non-existant subfields. I added a guard clause to the subfield and department term-from-id functions to prevent these bugs and made PR #1794 for it.
